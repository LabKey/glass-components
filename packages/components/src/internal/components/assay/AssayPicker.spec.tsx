import React from 'react';
import {mount} from 'enzyme';
import {AssayPicker, AssayPickerTabs} from '../../..';

import renderer from "react-test-renderer";
import {Simulate} from "react-dom/test-utils";
import click = Simulate.click;

describe('AssayPicker', () => {
    test('AssayPicker', () => {
        const component = <AssayPicker
            showImport={true}
            onChange={jest.fn()}
        />;

        const wrapper = mount(component);

        // Verify all three tabs shown and standard assay selected
        expect(wrapper.find('.nav-tabs li')).toHaveLength(3);
        expect(wrapper.find('.nav-tabs li.active a#assay-picker-tabs-tab-standard')).toHaveLength(1);
        expect(wrapper.find('#assay-type-select-container')).toHaveLength(2);

        // Click import tab and verify it's shown
        expect(wrapper.find('.nav-tabs li.active a#assay-picker-tabs-tab-import')).toHaveLength(0);
        const importTab = wrapper.find('.nav-tabs li a#assay-picker-tabs-tab-import');
        expect(importTab).toHaveLength(1);
        importTab.simulate("click");
        expect(wrapper.find('.nav-tabs li.active a#assay-picker-tabs-tab-import')).toHaveLength(1);
        expect(wrapper.find('.file-upload--container')).toHaveLength(1);

        expect(wrapper).toMatchSnapshot();
        wrapper.unmount();
    });

    test('AssayPicker No Import, Selected Specialty tab', () => {
        const component = <AssayPicker
            showImport={false}
            selectedTab={AssayPickerTabs.SPECIALTY_ASSAY_TAB}
            onChange={jest.fn()}
        />;

        const wrapper = mount(component);

        // Verify only two tabs and specialty tab selected
        expect(wrapper.find('.nav-tabs li')).toHaveLength(2);
        expect(wrapper.find('.nav-tabs li.active a#assay-picker-tabs-tab-specialty')).toHaveLength(1);
        expect(wrapper.find('select')).toHaveLength(2);

        wrapper.unmount();
    });


});
