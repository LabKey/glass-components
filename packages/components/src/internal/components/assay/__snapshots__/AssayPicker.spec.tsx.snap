// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssayPicker AssayPicker 1`] = `
<Memo()
  onChange={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "container": "",
            "file": undefined,
            "provider": undefined,
            "tab": undefined,
          },
        ],
        Array [
          Object {
            "container": "",
            "file": undefined,
            "provider": undefined,
            "tab": "standard",
          },
        ],
        Array [
          Object {
            "container": "",
            "file": undefined,
            "provider": undefined,
            "tab": "import",
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  showImport={true}
>
  <div>
    <ForwardRef
      activeKey="import"
      defaultActiveKey="standard"
      id="assay-picker-tabs"
      onSelect={[Function]}
    >
      <Uncontrolled(TabContainer)
        activeKey="import"
        defaultActiveKey="standard"
        id="assay-picker-tabs"
        innerRef={null}
        onSelect={[Function]}
      >
        <TabContainer
          activeKey="import"
          id="assay-picker-tabs"
          onSelect={[Function]}
        >
          <Row
            bsClass="row"
            className="clearfix"
            componentClass="div"
            id="assay-picker-tabs"
          >
            <div
              className="clearfix row"
              id="assay-picker-tabs"
            >
              <Col
                bsClass="col"
                componentClass="div"
                sm={12}
              >
                <div
                  className="col-sm-12"
                >
                  <Nav
                    bsClass="nav"
                    bsStyle="tabs"
                    justified={false}
                    pullLeft={false}
                    pullRight={false}
                    stacked={false}
                  >
                    <ul
                      className="nav nav-tabs"
                      role="tablist"
                    >
                      <NavItem
                        active={false}
                        activeKey="import"
                        aria-controls="assay-picker-tabs-pane-standard"
                        disabled={false}
                        eventKey="standard"
                        id="assay-picker-tabs-tab-standard"
                        key=".0"
                        onKeyDown={[Function]}
                        onSelect={[Function]}
                        role="tab"
                        tabIndex={-1}
                      >
                        <li
                          className=""
                          role="presentation"
                        >
                          <SafeAnchor
                            aria-controls="assay-picker-tabs-pane-standard"
                            aria-selected={false}
                            componentClass="a"
                            disabled={false}
                            id="assay-picker-tabs-tab-standard"
                            onClick={[Function]}
                            onKeyDown={[Function]}
                            role="tab"
                            tabIndex={-1}
                          >
                            <a
                              aria-controls="assay-picker-tabs-pane-standard"
                              aria-selected={false}
                              href="#"
                              id="assay-picker-tabs-tab-standard"
                              onClick={[Function]}
                              onKeyDown={[Function]}
                              role="tab"
                              tabIndex={-1}
                            >
                              Standard Assay
                            </a>
                          </SafeAnchor>
                        </li>
                      </NavItem>
                      <NavItem
                        active={false}
                        activeKey="import"
                        aria-controls="assay-picker-tabs-pane-specialty"
                        disabled={false}
                        eventKey="specialty"
                        id="assay-picker-tabs-tab-specialty"
                        key=".1"
                        onKeyDown={[Function]}
                        onSelect={[Function]}
                        role="tab"
                        tabIndex={-1}
                      >
                        <li
                          className=""
                          role="presentation"
                        >
                          <SafeAnchor
                            aria-controls="assay-picker-tabs-pane-specialty"
                            aria-selected={false}
                            componentClass="a"
                            disabled={false}
                            id="assay-picker-tabs-tab-specialty"
                            onClick={[Function]}
                            onKeyDown={[Function]}
                            role="tab"
                            tabIndex={-1}
                          >
                            <a
                              aria-controls="assay-picker-tabs-pane-specialty"
                              aria-selected={false}
                              href="#"
                              id="assay-picker-tabs-tab-specialty"
                              onClick={[Function]}
                              onKeyDown={[Function]}
                              role="tab"
                              tabIndex={-1}
                            >
                              Specialty Assays
                            </a>
                          </SafeAnchor>
                        </li>
                      </NavItem>
                      <NavItem
                        active={true}
                        activeKey="import"
                        aria-controls="assay-picker-tabs-pane-import"
                        disabled={false}
                        eventKey="import"
                        id="assay-picker-tabs-tab-import"
                        key=".2"
                        onKeyDown={[Function]}
                        onSelect={[Function]}
                        role="tab"
                      >
                        <li
                          className="active"
                          role="presentation"
                        >
                          <SafeAnchor
                            aria-controls="assay-picker-tabs-pane-import"
                            aria-selected={true}
                            componentClass="a"
                            disabled={false}
                            id="assay-picker-tabs-tab-import"
                            onClick={[Function]}
                            onKeyDown={[Function]}
                            role="tab"
                          >
                            <a
                              aria-controls="assay-picker-tabs-pane-import"
                              aria-selected={true}
                              href="#"
                              id="assay-picker-tabs-tab-import"
                              onClick={[Function]}
                              onKeyDown={[Function]}
                              role="tab"
                            >
                              Import Assay Design
                            </a>
                          </SafeAnchor>
                        </li>
                      </NavItem>
                    </ul>
                  </Nav>
                </div>
              </Col>
              <Col
                bsClass="col"
                componentClass="div"
                sm={12}
              >
                <div
                  className="col-sm-12"
                >
                  <TabContent
                    animation={true}
                    bsClass="tab"
                    componentClass="div"
                    mountOnEnter={false}
                    unmountOnExit={false}
                  >
                    <div
                      className="tab-content"
                    >
                      <TabPane
                        bsClass="tab-pane"
                        className="margin-bottom margin-top"
                        eventKey="standard"
                      >
                        <Fade
                          appear={false}
                          in={false}
                          mountOnEnter={false}
                          onEnter={[Function]}
                          onExited={[Function]}
                          timeout={300}
                          unmountOnExit={false}
                        >
                          <Transition
                            appear={false}
                            enter={true}
                            exit={true}
                            in={false}
                            mountOnEnter={false}
                            onEnter={[Function]}
                            onEntered={[Function]}
                            onEntering={[Function]}
                            onExit={[Function]}
                            onExited={[Function]}
                            onExiting={[Function]}
                            timeout={300}
                            unmountOnExit={false}
                          >
                            <div
                              aria-hidden={false}
                              aria-labelledby="assay-picker-tabs-tab-standard"
                              className="fade margin-bottom margin-top tab-pane active"
                              id="assay-picker-tabs-pane-standard"
                              role="tabpanel"
                            >
                              <Memo()>
                                <div>
                                  <Row
                                    bsClass="row"
                                    componentClass="div"
                                  >
                                    <div
                                      className="row"
                                    >
                                      <Col
                                        bsClass="col"
                                        componentClass="div"
                                        xs={6}
                                      >
                                        <div
                                          className="col-xs-6"
                                        >
                                          <div
                                            className="margin-bottom"
                                          >
                                            <b>
                                              Standard Assay
                                            </b>
                                            <span
                                              className="gray-text"
                                            >
                                               (Recommended)
                                            </span>
                                          </div>
                                          <p>
                                            Standard assays are the most flexible choice for working with experimental data. Use this assay type to customize the format for mapping your experimental results.
                                          </p>
                                        </div>
                                      </Col>
                                    </div>
                                  </Row>
                                  <Row
                                    bsClass="row"
                                    componentClass="div"
                                  >
                                    <div
                                      className="row"
                                    >
                                      <Col
                                        bsClass="col"
                                        componentClass="div"
                                        xs={6}
                                      >
                                        <div
                                          className="col-xs-6"
                                        >
                                          <div
                                            className="margin-bottom margin-top"
                                          >
                                            <b>
                                              Supported File Types
                                            </b>
                                          </div>
                                          <p />
                                        </div>
                                      </Col>
                                    </div>
                                  </Row>
                                  <div
                                    className="margin-top"
                                  >
                                    <Row
                                      bsClass="row"
                                      componentClass="div"
                                    >
                                      <div
                                        className="row"
                                      >
                                        <Col
                                          bsClass="col"
                                          componentClass="div"
                                          xs={6}
                                        >
                                          <div
                                            className="col-xs-6"
                                          >
                                            <Memo()
                                              onChange={[Function]}
                                              selected=""
                                            >
                                              <div
                                                className="margin-bottom"
                                              >
                                                <b>
                                                  Assay Location
                                                </b>
                                              </div>
                                              <p>
                                                Choose where the assay will be used and visible within subfolders.
                                              </p>
                                              <select
                                                className="form-control"
                                                id="assay-type-select-container"
                                                onChange={[Function]}
                                                value=""
                                              />
                                            </Memo()>
                                          </div>
                                        </Col>
                                      </div>
                                    </Row>
                                  </div>
                                </div>
                              </Memo()>
                            </div>
                          </Transition>
                        </Fade>
                      </TabPane>
                      <TabPane
                        bsClass="tab-pane"
                        className="margin-bottom margin-top"
                        eventKey="specialty"
                      >
                        <Fade
                          appear={false}
                          in={false}
                          mountOnEnter={false}
                          onEnter={[Function]}
                          onExited={[Function]}
                          timeout={300}
                          unmountOnExit={false}
                        >
                          <Transition
                            appear={false}
                            enter={true}
                            exit={true}
                            in={false}
                            mountOnEnter={false}
                            onEnter={[Function]}
                            onEntered={[Function]}
                            onEntering={[Function]}
                            onExit={[Function]}
                            onExited={[Function]}
                            onExiting={[Function]}
                            timeout={300}
                            unmountOnExit={false}
                          >
                            <div
                              aria-hidden={true}
                              aria-labelledby="assay-picker-tabs-tab-specialty"
                              className="fade margin-bottom margin-top tab-pane"
                              id="assay-picker-tabs-pane-specialty"
                              role="tabpanel"
                            >
                              <Memo()
                                onChange={[Function]}
                              >
                                <div>
                                  <Row
                                    bsClass="row"
                                    componentClass="div"
                                  >
                                    <div
                                      className="row"
                                    >
                                      <Col
                                        bsClass="col"
                                        componentClass="div"
                                        xs={6}
                                      >
                                        <div
                                          className="col-xs-6"
                                        >
                                          <div
                                            className="margin-bottom"
                                          >
                                            <b>
                                              Use Instrument Specific Data Format
                                            </b>
                                          </div>
                                          <div
                                            className="margin-bottom"
                                          />
                                          <p>
                                            <div
                                              dangerouslySetInnerHTML={
                                                Object {
                                                  "__html": undefined,
                                                }
                                              }
                                            />
                                          </p>
                                        </div>
                                      </Col>
                                    </div>
                                  </Row>
                                  <Row
                                    bsClass="row"
                                    componentClass="div"
                                  >
                                    <div
                                      className="row"
                                    >
                                      <Col
                                        bsClass="col"
                                        componentClass="div"
                                        xs={6}
                                      >
                                        <div
                                          className="col-xs-6"
                                        >
                                          <div
                                            className="margin-top margin-bottom"
                                          >
                                            <b>
                                              Supported File Types
                                            </b>
                                          </div>
                                          <p />
                                        </div>
                                      </Col>
                                    </div>
                                  </Row>
                                  <div
                                    className="margin-top"
                                  >
                                    <Row
                                      bsClass="row"
                                      componentClass="div"
                                    >
                                      <div
                                        className="row"
                                      >
                                        <Col
                                          bsClass="col"
                                          componentClass="div"
                                          xs={6}
                                        >
                                          <div
                                            className="col-xs-6"
                                          >
                                            <Memo()
                                              onChange={[Function]}
                                              selected=""
                                            >
                                              <div
                                                className="margin-bottom"
                                              >
                                                <b>
                                                  Assay Location
                                                </b>
                                              </div>
                                              <p>
                                                Choose where the assay will be used and visible within subfolders.
                                              </p>
                                              <select
                                                className="form-control"
                                                id="assay-type-select-container"
                                                onChange={[Function]}
                                                value=""
                                              />
                                            </Memo()>
                                          </div>
                                        </Col>
                                      </div>
                                    </Row>
                                  </div>
                                </div>
                              </Memo()>
                            </div>
                          </Transition>
                        </Fade>
                      </TabPane>
                      <TabPane
                        bsClass="tab-pane"
                        className="margin-bottom margin-top"
                        eventKey="import"
                      >
                        <Fade
                          appear={false}
                          in={false}
                          mountOnEnter={false}
                          onEnter={[Function]}
                          onExited={[Function]}
                          timeout={300}
                          unmountOnExit={false}
                        >
                          <Transition
                            appear={false}
                            enter={true}
                            exit={true}
                            in={false}
                            mountOnEnter={false}
                            onEnter={[Function]}
                            onEntered={[Function]}
                            onEntering={[Function]}
                            onExit={[Function]}
                            onExited={[Function]}
                            onExiting={[Function]}
                            timeout={300}
                            unmountOnExit={false}
                          >
                            <div
                              aria-hidden={true}
                              aria-labelledby="assay-picker-tabs-tab-import"
                              className="fade margin-bottom margin-top tab-pane"
                              id="assay-picker-tabs-pane-import"
                              role="tabpanel"
                            >
                              <Memo()
                                onFileChange={[Function]}
                                onFileRemove={[Function]}
                              >
                                <div>
                                  <Row
                                    bsClass="row"
                                    componentClass="div"
                                  >
                                    <div
                                      className="row"
                                    >
                                      <Col
                                        bsClass="col"
                                        componentClass="div"
                                        xs={6}
                                      >
                                        <div
                                          className="col-xs-6"
                                        >
                                          <div
                                            className="margin-bottom"
                                          >
                                            <b>
                                              Import .XAR or .XAR.XML file
                                            </b>
                                          </div>
                                          <p>
                                            To create an assay from an existing design, import a
                                             
                                            <a
                                              href="https://www.labkey.org/Documentation/wiki-page.view?name=XarTutorial"
                                            >
                                              .XAR or .XAR.XML
                                            </a>
                                            file here. You will be able to customize the design after importing.
                                          </p>
                                        </div>
                                      </Col>
                                    </div>
                                  </Row>
                                  <Row
                                    bsClass="row"
                                    componentClass="div"
                                  >
                                    <div
                                      className="row"
                                    >
                                      <Col
                                        bsClass="col"
                                        componentClass="div"
                                        xs={12}
                                      >
                                        <div
                                          className="col-xs-12"
                                        >
                                          <FileAttachmentForm
                                            acceptedFormats=".XAR, .XAR.XML"
                                            allowDirectories={false}
                                            allowMultiple={false}
                                            cancelText="Cancel"
                                            compact={false}
                                            label="Attachments"
                                            labelLong="Select file or drag and drop here."
                                            onFileChange={[Function]}
                                            onFileRemoval={[Function]}
                                            showAcceptedFormats={true}
                                            showButtons={false}
                                            showLabel={false}
                                            showProgressBar={false}
                                            submitText="Upload"
                                          >
                                            <span
                                              className="translator--toggle__wizard"
                                            >
                                              <FormSection
                                                iconSpacer={false}
                                                label="Attachments"
                                                showLabel={false}
                                              >
                                                <div
                                                  className="row"
                                                >
                                                  <div
                                                    className="col-sm-12"
                                                  >
                                                    <div
                                                      className="wizard-row--container"
                                                    >
                                                      <div
                                                        className=""
                                                      >
                                                        <FileAttachmentContainer
                                                          acceptedFormats=".XAR, .XAR.XML"
                                                          allowDirectories={false}
                                                          allowMultiple={false}
                                                          compact={false}
                                                          handleChange={[Function]}
                                                          handleRemoval={[Function]}
                                                          labelLong="Select file or drag and drop here."
                                                        >
                                                          <div
                                                            className="file-upload--container block"
                                                          >
                                                            <label
                                                              className="file-upload--label"
                                                              htmlFor="fileUpload"
                                                              onDragEnter={[Function]}
                                                              onDragLeave={[Function]}
                                                              onDragOver={[Function]}
                                                              onDrop={[Function]}
                                                            >
                                                              <i
                                                                aria-hidden="true"
                                                                className="fa fa-cloud-upload fa-2x cloud-logo"
                                                              />
                                                              Select file or drag and drop here.
                                                            </label>
                                                            <input
                                                              accept=".XAR, .XAR.XML"
                                                              className="file-upload--input"
                                                              id="fileUpload"
                                                              multiple={false}
                                                              name="fileUpload"
                                                              onChange={[Function]}
                                                              type="file"
                                                            />
                                                          </div>
                                                        </FileAttachmentContainer>
                                                      </div>
                                                    </div>
                                                  </div>
                                                </div>
                                              </FormSection>
                                            </span>
                                            <div
                                              className="row"
                                            >
                                              <div
                                                className="col-md-9"
                                              >
                                                <div
                                                  className="file-form-formats"
                                                >
                                                  <strong>
                                                    Supported formats include: 
                                                  </strong>
                                                  .XAR, .XAR.XML
                                                </div>
                                              </div>
                                              <div
                                                className="col-md-3"
                                              >
                                                <div
                                                  className="pull-right"
                                                />
                                              </div>
                                            </div>
                                          </FileAttachmentForm>
                                        </div>
                                      </Col>
                                    </div>
                                  </Row>
                                  <Row
                                    bsClass="row"
                                    componentClass="div"
                                  >
                                    <div
                                      className="row"
                                    >
                                      <Col
                                        bsClass="col"
                                        componentClass="div"
                                        xs={6}
                                      >
                                        <div
                                          className="col-xs-6"
                                        >
                                          <div
                                            className="margin-top margin-bottom"
                                          >
                                            <b>
                                              Import through data pipeline
                                            </b>
                                          </div>
                                          <div
                                            className="margin-top margin-bottom"
                                          >
                                            <p>
                                              If you have an existing assay .XAR file on this server, you can directly upload this assay design using this folder's data pipeline.
                                            </p>
                                          </div>
                                          <div
                                            className="margin-top"
                                          >
                                            <p>
                                              <a
                                                href="/labkey/pipeline/browse.view"
                                              >
                                                Use data pipeline
                                              </a>
                                            </p>
                                          </div>
                                        </div>
                                      </Col>
                                    </div>
                                  </Row>
                                </div>
                              </Memo()>
                            </div>
                          </Transition>
                        </Fade>
                      </TabPane>
                    </div>
                  </TabContent>
                </div>
              </Col>
            </div>
          </Row>
        </TabContainer>
      </Uncontrolled(TabContainer)>
    </ForwardRef>
  </div>
</Memo()>
`;
